runtime/lua/_vim9script.lua

```diff
@@ -0,0 +1,634 @@
+  endtry
+
+  return [ret, a:args]
+endfunction
+]])
+
+NVIM9['autoload'] = (function()
+  return function(path)
+    return loadfile(path)()
+  end
+end)()
+NVIM9['bool'] = (function()
+  return function(...)
+    return NVIM9.convert.to_vim_bool(...)
+  end
+end)()
+NVIM9['convert'] = (function()
+  local M = {}
+
+  M.decl_bool = function(val)
+    if type(val) == 'boolean' then
+      return val
+    elseif type(val) == 'number' then
+      if val == 0 then
+        return false
+      elseif val == 1 then
+        return true
+      else
+        error(string.format('bad number passed to bool declaration: %s', val))
+      end
+    end
+
+    error(string.format('invalid bool declaration: %s', vim.inspect(val)))
+  end
+
+  M.decl_dict = function(val)
+    if type(val) == 'nil' then
+      return vim.empty_dict()
+    elseif type(val) == 'table' then
+      if vim.tbl_isempty(val) then
+        return vim.empty_dict()
+      elseif vim.tbl_islist(val) then
+        error(string.format('Cannot pass list to dictionary? %s', vim.inspect(val)))
+      else
+        return val
+      end
+    end
+
+    error(string.format('invalid dict declaration: %s', vim.inspect(val)))
+  end
+
+  M.to_vim_bool = function(val)
+    if type(val) == 'boolean' then
+      return val
+    elseif type(val) == 'number' then
+      return val ~= 0
+    elseif type(val) == 'string' then
+      return string.len(val) ~= 0
+    elseif type(val) == 'table' then
+      return not vim.tbl_isempty(val)
+    elseif val == nil then
+      return false
+    end
+
+    error('unhandled type: ' .. vim.inspect(val))
+  end
+
+  return M
+end)()
+NVIM9['fn'] = (function()
+  local M = {}
+
+  M.insert = function(list, item, idx)
+    if idx == nil then
+      idx = 1
+    end
+
+    table.insert(list, idx + 1, item)
+
+    return list
+  end
+
+  M.extend = function(left, right, expr3)
+    if expr3 ~= nil then
+      error("haven't written this code yet")
+    end
+
+    if vim.tbl_islist(right) then
+      vim.list_extend(left, right)
+      return left
+    else
+      -- local result = vim.tbl_extend(left, right)
+      for k, v in pairs(right) do
+        left[k] = v
+      end
+
+      return left
+    end
+  end
+
+  M.add = function(list, item)
+    table.insert(list, item)
+    return list
+  end
+
+  M.has_key = function(obj, key)
+    return not not obj[key]
+  end
+
+  M.prop_type_add = function(...)
+    local args = { ... }
+    print('[prop_type_add]', vim.inspect(args))
+  end
+
+  do
+    local patch_overrides = {
+      -- We do have vim9script :) that's this plugin
+      ['vim9script'] = true,
+
+      -- Include some vim patches that I don't care about
+      [ [[patch-8.2.2261]] ] = true,
+      [ [[patch-8.2.4257]] ] = true,
+    }
+
+    M.has = function(patch)
+      if patch_overrides[patch] then
+        return true
+      end
+
+      return vim.fn.has(patch)
+    end
+  end
+
+  --[=[
+
+
+readdirex({directory} [, {expr} [, {dict}]])			*readdirex()*
```

Comment:
I guess this is a vim-patch we're missing?

Reply:
Yes. It depends on a lot of platform-specific function in Vim.